### DISCUSSION TITLE
Looks promising, but have some questions

### DISCUSSION THREAD

**Discussion by mvdwetering** [petchmakes/ESP32-IRBlaster-UCR2#3] [General]
Hi, this looks promising and might be able to solve my annoyance with having the provided IR emitters not being powerful enough and having another cable going to my dock which is not really pretty.

I was wondering, would this solution be able to support the full power of the Logitech Harmony blasters? The UC IR emitters can't get to all my devices so I now trigger scripts in HA which in turn use the Harmony blasters through the hub. Would be nice to get rid of that hack.

Maybe a random question, but did you consider making this an [Esphome external component](https://esphome.io/components/external_components)?
That way you could focus on the UC dock API part and let Esphome handle things like OTA, Wifi/Ethernet setup and basic webpage. Also things like pin assignments could be easily changed through the configuration instead of having to edit the code. That being said I have no experience with developing for Esphome, so no clue how easy/suitable it really is.

On a sidenote, I saw a screenshot on the UC discord from the webpage of this project and noticed there was a UC logo on it. You should probably not use that logo for non UC products.

**Comment by petchmakes** [comment]
> I was wondering, would this solution be able to support the full power of the Logitech Harmony blasters? The UC IR emitters can't get to all my devices so I now trigger scripts in HA which in turn use the Harmony blasters through the hub. Would be nice to get rid of that hack.

I would expect the IR power to be able to match the harmony. Though it would depend on the number and type of emitters. I assumed 5 of the higher current emitters as spec'd int he schematic is enough but need to test. Just add more if required (so long as power source and transistor can handle). Maybe adding  capacitor across the supply rails to add a bit of extra capacity to the power supply for IR bursts, but shouldn't be necessary with reasonable PSU.

> Maybe a random question, but did you consider making this an [Esphome external component](https://esphome.io/components/external_components)?

Would be nice - things like webconfig, home assistant config, etc.  The approach I took was the lowest barrier of entry for me (from a development perspective). But I think it would be neat to try to get it working under esphome, someone is welcome to try or maybe I'll take a look at some point. It should be possible if esphome allows the ir send task to run on it's own core (this is very time sensitive).

> On a sidenote, I saw a screenshot on the UC discord from the webpage of this project and noticed there was a UC logo on it. You should probably not use that logo for non UC products.

Agree. I've ask for this to be removed in the pull request.

**Reply by mvdwetering** [reply]
>I assumed 5 of the higher current emitters as spec'd int he schematic is enough but need to test. Just add more if required (so long as power source and transistor can handle

I was more wondering if I could just connect the Harmony blasters to the mini jack connectors.

>It should be possible if esphome allows the ir send task to run on it's own core (this is very time sensitive).

I assume Esphome should be able to handle sending IR since they have a [RemoteTransmitter  component](https://esphome.io/components/remote_transmitter.html) that allows for IR transmission. The page has this note about performance.

-- This component is more accurate on the ESP32, since that chipset has a dedicated peripheral for sending exact signal sequences.

**Reply by petchmakes** [reply]
> I was more wondering if I could just connect the Harmony blasters to the mini jack connectors.

It should work. It's a diode, so orientation matters, I've specified it to match the UC emitters, i.e. the tip positive. I guess there's a small risk of overloading them, not sure what current they can take. Could try inserting a low ohm resistor in series first.

> I assume Esphome should be able to handle sending IR since they have a [RemoteTransmitter component](https://esphome.io/components/remote_transmitter.html) that allows for IR transmission. The page has this note about performance.

>
> -- This component is more accurate on the ESP32, since that chipset has a dedicated peripheral for sending exact signal sequences.

I'm using the same library (or at least the library forked in Unfolded Circles github account - IRremoteESP8266). Thought it made most sense for consistent behaviour. May be worth trying out if someone wants to try an esphome based implementation. I won't have time at the moment.
